### PR TITLE
Fix the String lastIndexOf() & indexOf() docs

### LINKED ISSUE #12351: Issue with "String.prototype.lastIndexOf()": lastIndexOf returning wrong response
MDN URL: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/lastIndexOf

#### Did you test this? If so, how?
`const testString = "locate string locate";`
`
testString.lastIndexOf('locate');`
`> 14`
`
testString.lastIndexOf('locate', 14);`

`> 14 - how it is possible?`

image

<details>
<summary>MDN Content page report details</summary>

* Folder: `en-us/web/javascript/reference/global_objects/string/lastindexof`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/lastIndexOf
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/javascript/reference/global_objects/string/lastindexof/index.md
* Last commit: https://github.com/mdn/content/commit/cf607d68522cd35ee7670782d3ee3a361eaef2e4
* Document last modified: 2021-07-20T18:47:57.000Z

</details>

### PR DESCRIPTION
**Pull request by sideshowbarker**
The existing wording in the lastIndexOf() doc misled at least one reader (see https://github.com/mdn/content/issues/12351) — because, while for `indexOf()`, it works fine to frame the behavior in terms of searching “from” a particular index (“fromIndex”), that framing doesn’t work well at all for `lastIndexOf()`.

---

So the most important parts of this change are:

- renaming the second argument to “position” rather than “fromIndex”
- removing the wording “searching backwards” from the `lastIndexOf()` doc
- removing mention of “starting” the search at a particular position

The changes to the `indexOf()` doc are less necessary but are somewhat just for preserving consistency/parallelism with the wording needed for the `lastIndexOf()` doc. But the result is wording that’s still just as clear as the existing wording.

Fixes https://github.com/mdn/content/issues/12351

### CODE REVIEW

**Comment by Andrew-Cottrell** [comment]
Why have "_a number_" for `indexOf` and "_an integer_" for `lastIndexOf`; could they use the same terminology?

Apart from that very minor aspect: looks good to me.

**Comment by sideshowbarker** [comment]
> Why have "_a number_" for `indexOf` and "_an integer_" for `lastIndexOf`; could they use the same terminology?

oops, thanks for catching that — fixed now (by changing it so that we have “number” in both articles)

File: files/en-us/web/javascript/reference/global_objects/string/indexof/index.md [hunk 1]
```diff
@@ -7,69 +7,45 @@ tags:
...
 
-    So, for example: `'undefined'.indexOf()` will return `0`, as
-    `undefined` is found at position `0` in the string
-    `undefined`. `'undefine'.indexOf()` however will return
-    `-1`, as `undefined` is not found in the string
-    `undefine`.
+- `position` {{optional_inline}}
 
-- _`fromIndex`_ {{optional_inline}}
+  - : The method returns the index of the first occurrence of the specified substring at a position greater than or equal to `position`, which defaults to `0`. If `position` is greater than the length of the calling string, the method behaves as if no `position` had been specified. If `position` is less than zero, the method behaves as it would if `position` were `0`.
```
Comment (wbamberg) [inline files/en-us/web/javascript/reference/global_objects/string/indexof/index.md:34]: Sorry if I'm being slow here, but this bit:

> If `position` is greater than the length of the calling string, the method behaves as if no `position` had been specified.

...seems to be incompatible with the example given just below:

```js
'hello world hello'.indexOf('o', 99) // -1
```

...this doesn't seem to behave the same as if `position` were not specified:

```js
'hello world hello'.indexOf('o') // 4
```

Comment (sideshowbarker) [reply files/en-us/web/javascript/reference/global_objects/string/indexof/index.md]: > Sorry if I'm being slow here, but this bit:
> 
> > If `position` is greater than the length of the calling string, the method behaves as if no `position` had been specified.
> 
> ...seems to be incompatible with the example

Thanks for catching that — that was indeed wrong. Fixed now

**Review by wbamberg — APPROVED** [review]
👍 thanks for this update, looks much better like this.

**Pull request merged**